### DISCUSSION TITLE
Inno-maker CAN - USB converter OSX

### DISCUSSION THREAD

**Discussion by minoseigenheer** [collin80/SavvyCAN#437] [Q&A]
Has anyone tried to get the Inno-maker CAN - USB converter working with SavvyCAN on OSX or Win?
https://www.inno-maker.com/product/usb-can/
https://github.com/INNO-MAKER/usb2can

**Comment by mickeyl** [comment]
While this is a great device for the use with Linux, to my knowledge no one wrote a libusb-derived user space driver yet. It should be relatively simple to derive the protocol by looking at the `gs_usb` socketcan kernel driver.

**Comment by Tomblarom** [comment]
Check the manual file: Page 21 "7. Mac Os API" in [USB2CAN UserManual v.1.7.pdf](https://github.com/INNO-MAKER/usb2can/files/7981795/USB2CAN.UserManual.v.1.7.pdf)

Can you do this for windows as well?

**Comment by KarlZeilhofer** [comment]
Hi, 
how do I use the Innomaker USB2CAN-X2 with SavvyCAN on Linux? 
I do see can0 and can1 devices in Wireshark listed. But how do I configure SavvyCAN on Linux with this device?
Greets, Karl

**Reply by KarlZeilhofer** [reply]
In the mean time I found it out. Run following lines to setup ```can0``` and ```can1```: 
```
sudo apt-get install can-utils
sudo modprobe can
sudo modprobe can_raw
sudo ip link set can0 type can bitrate 500000
sudo ip link set up can0
sudo ip link set can1 type can bitrate 500000
sudo ip link set up can1
```
In SavvyCAN (V208): 
```
		Menü --> Connection --> Open Connection Window
			Add new device connection
			QT Serial Bus
			socketcan
			can0/can1
```

**Comment by GPSdualSport** [comment]
My experience with the Inno-maker CAN - USB converter was not good. Running Ubuntu 22.04 the device seemed to work fine with initial set up using Can-util.
Connecting to my ISO 157765-4 ECU only returned two PIDs. Contacted Inno-Maker about the issue Calvin quickly responded with advice that did not address the issue. My second attempt included candump results and more details on initial set up results. I also included info on the fact my $18 ELM327 bluetooth adapter and Android App has no issues reading the ODBII data on the ECU.
Calvin had no answers on why this was happening. So the Inno-maker CAN - USB converter got returned to Amazon.
SavvyCan did connect with the Inno-maker CAN - USB converter with no issues. But only the same two PIDs appeared.
